### DISCUSSION TITLE
Is it possible to specify separate extension icons for light/dark VSCode themes?

### DISCUSSION THREAD

**Discussion by ymc9** [microsoft/vscode-discussions#220] [Extension Development QnA] [answered]
Hi there,

I see that in package.json of an extension, I can set light/dark icons for my language service:
```json
"contributes": {
        "languages": [
            {
                ...
                "icon": {
                    "light": "./asset/logo-light-256.png",
                    "dark": "./asset/logo-dark-256.png"
                }
            }
        ]
    }
```
However, I can't find how to do this for the entire extension. The top-level "icon" setting seems to only accept one single string.

Without this support, grayscale icons can look pretty bad on light/dark themes.

**Comment by isidorn** [comment, marked as answer]
You need to do it per each icon.

**Reply by ymc9** [reply]
Sorry, I didn’t get it. I already have two icons, but not sure how to configure it in package.json. The “icon” field at toplevel seems to only accept a simple string.

**Reply by isidorn** [reply]
Aaa I misunderstood. You want the light / dark for the icon of your extension. That is currently not supported by the Marketplace.
fyi @prashantvc

**Reply by ymc9** [reply]
Just to make it more visual, my extension looks like this when VSCode is in a dark theme 😂:

<img width="969" alt="image" src="https://user-images.githubusercontent.com/104139426/200719756-429bae87-726a-41c5-81d7-d5fbab56dc15.png">
